src/unit/unit.cpp

```diff
@@ -2671,14 +2862,17 @@ static void HitUnit_AttackBack(CUnit &attacker, CUnit &target)
 	COrder *savedOrder = NULL;
 
 	if (target.Player->AiEnabled == false) {
-		if (target.CurrentAction() == UnitActionAttack) {
-			COrder_Attack &order = dynamic_cast<COrder_Attack &>(*target.CurrentOrder());
-			if (order.IsWeakTargetSelected() == false) {
-				return;
-			}
-		} else {
-			return;
-		}
+		return;
+		/*
```

Comment:
This is commented out because the unit will now automatically select the right target, and we don't need to react to this event anymore, right?

Reply:
In general yes, but I supposed there are some rare and certain situations when we will have to react to this event (f.e. there is no targets/or only non-threat target in visible area, someone under fog of war attack this unit). Because of that this code was just commented instead of removing. But as all worked fine I just forgot about this part.
Added to my ToDo list.